src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processors/hmcts/CourtCaseEventsProcessor.kt

```diff
@@ -88,6 +91,16 @@ class CourtCaseEventsProcessor(
       ),
     )
 
+    val personEntities: List<PersonEntity> = personService.findCandidateRecords(person)
+    telemetryService.trackEvent(
+      TelemetryEventType.CPR_CANDIDATE_RECORD_SEARCH,
+      mapOf(
+        EventKeys.SOURCE_SYSTEM to SourceSystemType.HMCTS.name,
+        EventKeys.RECORD_COUNT to personEntities.size.toString(),
+        EventKeys.EVENT_TYPE to LIBRA_COURT_CASE.name,
+        EventKeys.MESSAGE_ID to sqsMessage.messageId,
+      ),
```

Comment:
I am not sure how best to do it, but we should put something in here which allows us to identify the matching algorithm we used
It will be hard to keep up to date.
Maybe a custom toString on the PersonSpecification??

Reply:
Under the hood it just constructs raw SQL from this it is very hard to determine which part of the statement did the matching part. So im not sure if this is possible with the way we have set it up??? we could log out the resulting SQL statement redacted...?

Reply:
You would have to override `hashcode` (and of course `equals`) to achieve this as the default would be the object reference and I don't think it should become a data class

Reply:
We need to be able to say "the average number of matches was 39 with algorithm 1, then we changed the levenshtein on date of birth to 1 and the average number went down to 5"

Reply:
Recording these rules in a confluence page for iteration purposes: https://dsdmoj.atlassian.net/wiki/spaces/PRD/pages/4938367021/Candidate+Searching+Algorithm